crates/biome_js_analyze/src/lint/nursery/use_explicit_function_return_type.rs

```diff
@@ -214,3 +230,30 @@ fn is_direct_const_assertion_in_arrow_functions(func: &AnyJsFunction) -> bool {
 
     ts_ref.text() == "const"
 }
+
+/// Checks if a function is allowed within specific expression contexts.
+/// These include function calls, array elements, and parenthesized expressions.
+///
+/// # Examples
+///
+/// JS_CALL_ARGUMENT_LIST:
+/// - `window.addEventListener('click', () => {});`
+/// - `const foo = arr.map(i => i * i);`
+/// - `setTimeout(function() { console.log("Hello!"); }, 1000);`
+///
+/// JS_ARRAY_ELEMENT_LIST:
+/// - `[function () {}, () => {}];`
+///
+/// JS_PARENTHESIZED_EXPRESSION:
+/// - `(function () {});`
+/// - `(() => {})();`
+fn is_function_used_in_argument_or_expression_list(func: &AnyJsFunction) -> bool {
+    matches!(
+        func.syntax().parent().kind(),
+        Some(
+            JsSyntaxKind::JS_CALL_ARGUMENT_LIST
+                | JsSyntaxKind::JS_ARRAY_ELEMENT_LIST
+                | JsSyntaxKind::JS_PARENTHESIZED_EXPRESSION
```

Comment:
I could add a comment just above to indicate that we have this for IIFE. We should also say that we assume that the parent of the parent is a call expression.

Reply:
comment added!